### ISSUE TITLE
Migration from Swing to JavaFX

### ISSUE THREAD

**Issue by IsaacMarovitz** [NationalSecurityAgency/ghidra#8109]
**Is your feature request related to a problem? Please describe.**
Ghidra's UI is a bit rough around the edges. It can be clunky, hard to learn for beginners, and lacks many accessibility and system integration features. For example, on macOS, Ghidra does not use the system menu bar or the system file importer. This was discussed back in 2019 in this issue (#830), with the conclusion mostly being that Swing was the major blocker.

**Describe the solution you'd like**
Migrating to Javafx would make the UI more maintainable, reduce boilerplate, and allow UI to be more clearly defined using FXML rather than purely declarative.

**Describe alternatives you've considered**
While Ghidra could continue to use Swing for the foreseeable future, it's likely to remain a blocker for future improvements to the UI. While moving to JavaFX is a significant undertaking, it would be worth it for the future. As I'm relatively new to Java as a whole, and Ghidra's codebase, I am curious to see if maintainers have any thoughts or further insights to share.
